config/v1/types_infrastructure.go

```diff
@@ -123,6 +123,20 @@ const (
 	ExternalTopologyMode TopologyMode = "External"
 )
 
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+// +kubebuilder:validation:Enum:="";"Internal";"External"
```

Comment:
> you can add tests
ack but before working on tests I want to make this is a valid API so I don't spend too much time in case this needs a major change.